cram/cram_io.c

```diff
@@ -4412,7 +4431,8 @@ void cram_free_slice(cram_slice *s) {
         int i;
 
         if (s->hdr) {
-            for (i = 0; i < s->hdr->num_blocks; i++) {
+            // minus naux_blocks as we free elsewhere?
```

Comment:
As far as I can tell, aux blocks are not freed elsewhere.  Although this does raise the question of if the data could be leaked between the pointers being [moved into `s->aux_block[]`](https://github.com/samtools/htslib/blob/be1630260dc374b5a82495d9394ac46149b2547c/cram/cram_encode.c#L2032) and then [copied into `s->block[]`](https://github.com/samtools/htslib/blob/be1630260dc374b5a82495d9394ac46149b2547c/cram/cram_encode.c#L1167) (called from [here](https://github.com/samtools/htslib/blob/be1630260dc374b5a82495d9394ac46149b2547c/cram/cram_encode.c#L2367))

Reply:
When reading the proof-of-concept files in #1697 and #1699, it looks like data is being leaked in this way.  It happens after failing to initialise a BETA codec [here](https://github.com/samtools/htslib/blob/be1630260dc374b5a82495d9394ac46149b2547c/cram/cram_encode.c#L2135).

Reply:
I did find the issue with BETA codec and fixed it, and the new fuzzer verifies this.  However oddly samtools still leaks.  Maybe we're still doing something different then.  I'll investigate.